src/main/cpp/src/hive_hash.cu

```diff
@@ -239,15 +483,20 @@ std::unique_ptr<cudf::column> hive_hash(cudf::table_view const& input,
   // Return early if there's nothing to hash
   if (input.num_columns() == 0 || input.num_rows() == 0) { return output; }
 
+  // Nested depth cannot exceed 8
+  constexpr int max_nested_depth = 8;
```

Comment:
Is this enough for general purposes? If not, we may want to consider another approach that preprocess the input so we can iterate the nested columns without using a stack.

Reply:
But how should list columns be handled without using a stack? 👀 
At least using `cudf::detail::lists_column_device_view(curr_col).get_sliced_child()` to flatten the list is not feasible. 

As illustrated in the example I provided in the comments, nulls at any nested level affect the result of `HiveHash`, and this information is lost during the flattening process.